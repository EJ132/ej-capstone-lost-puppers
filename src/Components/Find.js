/* eslint-disable array-callback-return */
import React, { Component } from 'react'
import '../Stylesheets/Main.css'
import '../Stylesheets/Find.css'
import '../Stylesheets/DogTags.css'
import DogTag from './LostDogs/Dogtag'
import config from '../config'
import NavBar from './Navbar'
import {Link} from 'react-router-dom'
import TokenService from '../services/token-service'

export default class Find extends Component {

    state = {
        dogTags: [],
        filter: null,
        filter_name: '',
        filter_zip: null,
        filter_zipVal: ''
    }

    filterResults = ev => {

        this.setState({
            filter: true,
            filter_name: ev.target.value
        })

        this.renderDogTags()

    }

    //need to implement when zipcode field is empty to set back to null or show all results

    filterByZip = ev => {

        this.setState({
            filter_zip: true,
            filter_zipVal: ev.target.value
        })

        this.renderDogTags()
    }

    clearFilter = () => {
        this.setState({
            filter: null,
            filter_name: '',
            filter_zip: null,
            filter_zipVal: ''
        })
    }

    componentDidMount() {
        fetch(`${config.API_ENDPOINT}/pups`, {method: 'GET'})
        .then(res =>
            (!res.ok)
              ? res.json().then(e => Promise.reject(e))
              : res.json()
        )
        .then(resJSON => {
            this.setState({
                dogTags: resJSON
            })
        })

        return;
    }

    handleDelete = cardId => {
        fetch(`${config.API_ENDPOINT}/pups/${cardId}`, {
        method: 'DELETE',
        headers: { 'authorization': `bearer ${TokenService.getAuthToken()}`}
        })
        .then(res => {
            this.setState({
                // eslint-disable-next-line eqeqeq
                dogTags: this.state.dogTags.filter(dog => dog.id != cardId)
            })
            return;
            })
    }

    renderDogTags() {

        let filteredResults = this.state.dogTags;

        if (this.state.filter) {
           // eslint-disable-next-line eqeqeq
           filteredResults = filteredResults.filter(dogCard => dogCard.category == this.state.filter_name)
        }

        // eslint-disable-next-line eqeqeq
        if (this.state.filter_zipVal == ''){
            return filteredResults.map(dogCard => {
                return <DogTag 
                name={dogCard.name}
                img={dogCard.image}
                description={dogCard.description}
                category={dogCard.category}
                dateCreated={dogCard.date_created}
                id={dogCard.id}
                key={dogCard.id}
                owner={dogCard.owner}
                delete={this.handleDelete}
                />
            });
        }

        if (this.state.filter_zip) {
            // eslint-disable-next-line eqeqeq
            filteredResults = filteredResults.filter(dogCard => dogCard.zipcode == this.state.filter_zipVal)
        }

        return filteredResults.map(dogCard => {
            return <DogTag 
            name={dogCard.name}
            img={dogCard.image}
            description={dogCard.description}
            category={dogCard.category}
            dateCreated={dogCard.date_created}
            id={dogCard.id}
            key={dogCard.id}
            owner={dogCard.owner}
            delete={this.handleDelete}
            />
        });
    }

    zipOptions = () => {

        const dogZips = this.state.dogTags.map(dog => dog.zipcode)
        const newArray = dogZips.filter((elem, i, arr) => {
            if (arr.indexOf(elem) === i) {
              return elem
            }
          })

        return (
        <select className="zipCodes" name='category' onChange={this.filterByZip}>
            {newArray.map(dog => {
             return <option key={dog} value={dog}>{dog}</option> })}
        </select>
    )}

    render() {
        return (
            <div>

                <NavBar />

                <div className="findContainer">
                    <div className="sidebar">
                        <h2>Sort</h2>
                            <div>
                                <form className="dogSizeFilter">
                                    <label>Dog Size</label>
                                    <div>
                                        <input type="radio" name="size" onChange={this.filterResults} value="Small" checked={this.state.filter_name === 'Small'}/>
                                        <label htmlFor="small">Small</label>
                                    </div>
                                    <div>
                                        <input type="radio" name="size" onChange={this.filterResults} value="Medium" checked={this.state.filter_name === 'Medium'}/>
                                        <label htmlFor="medium">Medium</label>
                                    </div>
                                    <div>
                                        <input type="radio" name="size" onChange={this.filterResults} value="Large" checked={this.state.filter_name === 'Large'}/>
                                        <label htmlFor="large">Large</label>
                                    </div>
                                    <div>
                                        <input type="button" id="filterButton" name="size" onClick={this.clearFilter} value="Clear"/>
                                    </div>
                                </form>
                            </div>
                            <div>
                                <label className="AreaCode">Area Code</label>
                                {this.zipOptions()}
                            </div>
                            <div className="addPup">
                                <Link to="/create">+ Add Pup</Link>
                            </div>
                    </div>

                    <div className="lostdogs">
                        <h2>Lost Pups</h2>

                        <div className="dogContainer">
                            {this.renderDogTags()}
                        </div>
                    </div>
                </div>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}



//implement automatic y axis for the messages