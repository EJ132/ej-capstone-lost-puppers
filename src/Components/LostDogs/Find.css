.findContainer {
  display: flex;
  flex-direction: row;
  padding-left: 10px;
  padding-right: 10px;
}

.sidebar {
  padding: 150px 20px 20px 20px;
  width: 18%;
  display: flex;
  flex-direction: column;
  color: white;
  text-align: center;
}

.dogSizeFilter {
  width: 100%;
  padding: 20px;
  display: flex;
  flex-direction: column;
}

.dogSizeFilter label {
  color: #79b877;
  display: inline-block;
  padding: 10px;
}

.AreaCode {
  font-weight: bold;
  font-size: 22px;
  display: block;
  padding: 20px;
}

.dogSizeFilter input {
  display: inline-block;
}

#filterButton {
  margin-top: 5px;
}

#areaCodeInput {
  text-align: center;
  padding: 5px;
  text-decoration: none;
}

.zipCodes {
  height: 30px;
  width: 80px;
  background-color: white;
  text-align: center;
  border-radius: 5px;
  padding: 7px;
}

.map {
  width: 100%;
  padding: 20px;
}

.map button {
  border: none;
  background: none;
  cursor: pointer;
}

@media screen and (max-width: 1270px) {
  .map {
    padding: 0;
    margin: 0;
    width: 100%;
    height: 100vh;
  }

  .findContainer {
    display: flex;
    flex-direction: column;
    padding-left: 10px;
    padding-right: 10px;
  }

  .sidebar {
    padding: 20px;
    width: 100%;
    display: flex;
    flex-direction: column;
    color: white;
    text-align: center;
    margin-top: 10%;
  }

  .dogSizeFilter {
    width: 100%;
    padding: 20px;
    display: flex;
    flex-direction: column;
  }

  .dogSizeFilter label {
    color: #79b877;
    display: inline-block;
    padding: 10px;
  }

  .AreaCode {
    font-weight: bold;
    font-size: 22px;
    display: block;
    padding: 20px;
  }

  .dogSizeFilter input {
    display: inline-block;
  }

  #filterButton {
    margin-top: 5px;
    background-color: black;
    border: 1px solid white;
    color: white;
    width: 40%;
    height: 30px;
  }

  #areaCodeInput {
    text-align: center;
    padding: 5px;
    text-decoration: none;
  }
}
