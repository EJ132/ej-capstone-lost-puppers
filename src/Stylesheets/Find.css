div > .findContainer {
    min-height: 120vh;
    height: auto;
}

.findContainer {
    display: flex;
    flex-direction: row;
    padding-left: 10px;
    padding-right: 10px;
}

.sidebar {
    padding: 20px;
    width: 20%;
    display: flex;
    flex-direction: column;
    color: white;
    text-align: center;
    margin-top: 15%;
}

.dogSizeFilter {
    width: 100%;
    padding: 20px;
    display: flex;
    flex-direction: column;
}

.dogSizeFilter label {
    color: orange;
    display: inline-block;
    padding: 10px;
}

.AreaCode {
    font-weight: bold;
    font-size: 22px;
    display: block;
    padding: 20px;
}

.dogSizeFilter input {
    display: inline-block;
}

#filterButton {
    margin-top: 5px;
}

#areaCodeInput {
    text-align: center;
    padding: 5px;
    text-decoration: none;
}

.lostdogs {
    padding: 20px;
    width: 80%;
    color: white;
}

.dogContainer {
    width: 100%;
    padding: 50px 10px 50px 10px;
    display: grid;
    justify-content: space-around;
    grid-template-columns: auto auto auto auto;
    grid-template-rows: 300px 300px;
}