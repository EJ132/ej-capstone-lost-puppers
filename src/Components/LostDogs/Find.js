/* eslint-disable array-callback-return */
import React, { Component } from 'react'
import '../MainPage/Main.css'
import './Find.css'
import './DogTags.css'
import DogTag from './Dogtag'
import NavBar from '../NavBar/Navbar'
import {Link} from 'react-router-dom'
import PupApiService from '../../services/thing-api-service'

export default class Find extends Component {

    state = {
        dogTags: [],
        filter: null,
        filter_name: '',
        filter_zip: null,
        filter_zipVal: ''
    }

    filterResults = ev => {

        this.setState({
            filter: true,
            filter_name: ev.target.value
        })

        this.renderDogTags()

    }

    filterByZip = ev => {

        this.setState({
            filter_zip: true,
            filter_zipVal: ev.target.value
        })

        this.renderDogTags()
    }

    clearFilter = () => {
        this.setState({
            filter: null,
            filter_name: '',
            filter_zip: null,
            filter_zipVal: ''
        })
    }

    componentDidMount() {
        PupApiService.getpups()
        .then(resJSON => {
            this.setState({
                dogTags: resJSON
            })
        })

        return;
    }

    renderDogTags() {

        let filteredResults = this.state.dogTags;

        if (this.state.filter) {
           // eslint-disable-next-line eqeqeq
           filteredResults = filteredResults.filter(dogCard => dogCard.category == this.state.filter_name)
        }

        // eslint-disable-next-line eqeqeq
        if (this.state.filter_zipVal == ''){
            return filteredResults.map(dogCard => {
                return <DogTag 
                name={dogCard.name}
                img={dogCard.image}
                description={dogCard.description}
                category={dogCard.category}
                dateCreated={dogCard.date_created}
                id={dogCard.id}
                key={dogCard.id}
                owner={dogCard.owner}
                />
            });
        }

        if (this.state.filter_zip) {
            // eslint-disable-next-line eqeqeq
            filteredResults = filteredResults.filter(dogCard => dogCard.zipcode == this.state.filter_zipVal)
        }

        return filteredResults.map(dogCard => {
            return <DogTag 
            name={dogCard.name}
            img={dogCard.image}
            description={dogCard.description}
            category={dogCard.category}
            dateCreated={dogCard.date_created}
            id={dogCard.id}
            key={dogCard.id}
            owner={dogCard.owner}
            />
        });
    }

    zipOptions = () => {

        const dogZips = this.state.dogTags.map(dog => dog.zipcode)
        const newArray = dogZips.filter((elem, i, arr) => {
            if (arr.indexOf(elem) === i) {
              return elem
            }
          })

        return (
        <select className="zipCodes" name='category' onChange={this.filterByZip}>
            {newArray.map(dog => {
             return <option key={dog} value={dog}>{dog}</option> })}
        </select>
    )}

    render() {
        return (
            <div>

                <NavBar />

                <div className="findContainer">
                    <div className="sidebar">
                        <h2>Sort</h2>
                            <div>
                                <form className="dogSizeFilter">
                                    <label>Dog Size</label>
                                    <div>
                                        <input type="radio" name="size" onChange={this.filterResults} value="Small" checked={this.state.filter_name === 'Small'}/>
                                        <label htmlFor="small">Small</label>
                                    </div>
                                    <div>
                                        <input type="radio" name="size" onChange={this.filterResults} value="Medium" checked={this.state.filter_name === 'Medium'}/>
                                        <label htmlFor="medium">Medium</label>
                                    </div>
                                    <div>
                                        <input type="radio" name="size" onChange={this.filterResults} value="Large" checked={this.state.filter_name === 'Large'}/>
                                        <label htmlFor="large">Large</label>
                                    </div>
                                    <div>
                                        <input type="button" id="filterButton" name="size" onClick={this.clearFilter} value="Clear"/>
                                    </div>
                                </form>
                            </div>
                            <div>
                                <label className="AreaCode">Area Code</label>
                                {this.zipOptions()}
                            </div>
                            <div className="addPup">
                                <Link to="/create">+ Add Pup</Link>
                            </div>
                    </div>

                    <div className="lostdogs">
                        <h2>Lost Pups</h2>

                        <div className="dogContainer">
                            {this.renderDogTags()}
                        </div>
                    </div>
                </div>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}



//implement automatic y axis for the messages