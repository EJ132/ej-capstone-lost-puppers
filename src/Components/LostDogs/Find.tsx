import * as React from "react";
import { Link } from "react-router-dom";

import PupApiService from "../../services/thing-api-service";
import NavBar from "../NavigationBar/NavigationBar";

import DogTag from "./Dogtag";
import Map from "./Map";

import "../Home/Home.css";
import "./Find.css";
import "./DogTags.css";

export default function Find() {
  const [dogTags, setDogTags] = React.useState([]);
  const [filter, setFilter] = React.useState(null);
  const [filterName, setFilterName] = React.useState("");
  const [filterZip, setFilterZip] = React.useState(null);
  const [filterZipValue, setFilterZipValue] = React.useState("");

  // CATEGORY FILTER
  function filterResults(ev: any) {
    const { value } = ev.target;

    setFilter(true);
    setFilterName(value);

    renderDogTags();
  }

  // ZIPCODE FILTER
  function filterByZip(ev: any) {
    const { value } = ev.target;

    setFilterZip(true);
    setFilterZipValue(value);

    renderDogTags();
  }

  // CLEARS FILTERS

  function clearFilter() {
    setFilter(false);
    setFilterName("");
    setFilterZip(false);
    setFilterZipValue("");
  }

  // FETCHES INFORMATION NEEDED TO RENDER THE DOGTAGS
  React.useEffect(() => {
    PupApiService.getpups().then((resJSON) => {
      setDogTags(resJSON);
    });
  }, []);

  // RENDERS OUT THE DOGS CARDS THAT WILL BE PASSED INTO MAP

  function renderDogTags() {
    let filteredResults = dogTags;

    if (filter) {
      filteredResults = filteredResults.filter(
        (dogCard) => dogCard.category === filterName
      );
    }

    if (!filterByZip) {
      return filteredResults.map((dogCard) => {
        return (
          <DogTag
            name={dogCard.name}
            img={dogCard.image}
            description={dogCard.description}
            category={dogCard.category}
            dateCreated={dogCard.date_created}
            lat={dogCard.lat}
            long={dogCard.long}
            id={dogCard.id}
            key={dogCard.id}
            owner={dogCard.owner}
          />
        );
      });
    }

    if (filterZip) {
      // eslint-disable-next-line eqeqeq
      filteredResults = filteredResults.filter(
        (dogCard) => dogCard.zipcode === filterZipValue
      );
    }

    return filteredResults.map((dogCard) => {
      return (
        <DogTag
          name={dogCard.name}
          img={dogCard.image}
          description={dogCard.description}
          category={dogCard.category}
          dateCreated={dogCard.date_created}
          lat={dogCard.lat}
          long={dogCard.long}
          id={dogCard.id}
          key={dogCard.id}
          owner={dogCard.owner}
        />
      );
    });
  }

  // MAKES SURE THAT ZIPCODES DONT REPEAT

  function zipOptions() {
    const dogZips = dogTags.map((dog) => dog.zipcode);
    const newArray = dogZips.filter((elem, i, arr) => {
      if (arr.indexOf(elem) === i) {
        return elem;
      }

      return null;
    });

    return (
      <select className="zipCodes" name="category" onChange={filterByZip}>
        {newArray.map((dog) => {
          return (
            <option key={dog} value={dog}>
              {dog}
            </option>
          );
        })}
      </select>
    );
  }

  return (
    <div>
      <NavBar />

      <div className="findContainer">
        <div className="sidebar">
          <h2>Sort</h2>
          <div>
            <form className="dogSizeFilter">
              <label>Dog Size</label>
              <div>
                <input
                  type="radio"
                  name="size"
                  onChange={filterResults}
                  value="Small"
                  checked={filterName === "Small"}
                />
                <label htmlFor="small">Small</label>
              </div>
              <div>
                <input
                  type="radio"
                  name="size"
                  onChange={filterResults}
                  value="Medium"
                  checked={filterName === "Medium"}
                />
                <label htmlFor="medium">Medium</label>
              </div>
              <div>
                <input
                  type="radio"
                  name="size"
                  onChange={filterResults}
                  value="Large"
                  checked={filterName === "Large"}
                />
                <label htmlFor="large">Large</label>
              </div>
              <div>
                <input
                  type="button"
                  id="filterButton"
                  name="size"
                  onClick={clearFilter}
                  value="Clear"
                />
              </div>
            </form>
          </div>
          <div>
            {/* <label className="AreaCode">Area Code</label> */}
            {zipOptions()}
          </div>
          <div className="addPup">
            <Link to="/create">+ Add Pup</Link>
          </div>
        </div>

        <div className="map">
          <Map dogTags={dogTags} />
        </div>
      </div>

      <footer>&#169; EJ Gonzalez 2019</footer>
    </div>
  );
}
