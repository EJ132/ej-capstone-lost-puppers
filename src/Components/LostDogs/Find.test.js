import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Find from './Find'

describe(`Find component`, () => {

  it('renders a Find page by default', () => {
    const wrapper = shallow(<Find />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
