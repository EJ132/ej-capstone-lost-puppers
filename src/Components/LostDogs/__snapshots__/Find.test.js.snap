// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find component renders a Find page by default 1`] = `
<div>
  <Navbar />
  <div
    className="findContainer"
  >
    <div
      className="sidebar"
    >
      <h2>
        Sort
      </h2>
      <div>
        <form
          className="dogSizeFilter"
        >
          <label>
            Dog Size
          </label>
          <div>
            <input
              checked={false}
              name="size"
              onChange={[Function]}
              type="radio"
              value="Small"
            />
            <label
              htmlFor="small"
            >
              Small
            </label>
          </div>
          <div>
            <input
              checked={false}
              name="size"
              onChange={[Function]}
              type="radio"
              value="Medium"
            />
            <label
              htmlFor="medium"
            >
              Medium
            </label>
          </div>
          <div>
            <input
              checked={false}
              name="size"
              onChange={[Function]}
              type="radio"
              value="Large"
            />
            <label
              htmlFor="large"
            >
              Large
            </label>
          </div>
          <div>
            <input
              id="filterButton"
              name="size"
              onClick={[Function]}
              type="button"
              value="Clear"
            />
          </div>
        </form>
      </div>
      <div>
        <label
          className="AreaCode"
        >
          Area Code
        </label>
        <select
          className="zipCodes"
          name="category"
          onChange={[Function]}
        />
      </div>
      <div
        className="addPup"
      >
        <Link
          to="/create"
        >
          + Add Pup
        </Link>
      </div>
    </div>
    <div
      className="lostdogs"
    >
      <h2>
        Lost Pups
      </h2>
      <div
        className="dogContainer"
      />
    </div>
  </div>
  <footer>
    Â© EJ Gonzalez 2019
  </footer>
</div>
`;
